Project Submission

AkshayaSrivastava opened the pull request:
I have a draft version in here entitled Final_Project_Draft. Please ignore this unless you would like to check something that i was working on in the process of completing the project. If needed, I can remove it as well, or file it in a separate folder for your convenience. 

The actual final project is entitled: "Telegrapher's Equations Final Project by Akshaya Srivastava"

Please use that for grading purposes.

ncclementi commented:
- Add license like the ones we have in our notebooks.
- You first paragraph is almost a copy form wikipedia, you need to re-write it using your own words, not just changing couple of them from the original source, that is technically plagiarism. 
- Before equation 1 you say: "From these Equations, the following model arises:" when you say "from these equations" to which ones are you referring? there are no equations before. 

*In the Voltage section, after "That is:" there should be a vertical space, between the text and the equation.

*In equations 21, 22, 23., 24 and so on. I would recomnd you to use \left[ and \right] in your
 LaTex so the brackets adjust to the size of what is it between them.
- When you want to import many modules from a library, you can just write:
  
  ```
      from matplotlib import pyplot, cm 
  ```
  
  you don't need to write them separately.
- If you have warnings, you should try to fixed them, not add a line to ignore them.
  
  ```
              warnings.filterwarnings('ignore')
  ```
  
  the problem is when you put a float as an index. Like here  v0[0:nx/2+1]=alpha,
  nx/2+1 this is a float.
- When you are plotting, if you add a semicolon in the last line of the cell,
  this:
         <matplotlib.legend.Legend at 0x7f3fdd068b70>
  
  won't appear and looksmore neat.
- You are using the same name (init) for 4 different functions, that might cause problems and /or confusion. 
- When you say you are "getting first order answers" what are you trying to say? I don't understand when you say that "it looks like a first order problem"
- Two of the "init" functions, are exactly the same. Once you define a function you don't need to write it again. You might have problems because you are using the same name 4 different times.
- In the In[19] the voltage appear to 1e37 which units are you using? Why this number? and it also appear the Initial in blue in the legend, but it isn't in the plot. 
- In the In[21] the legend says Initial, Neumann, Solved, but you only have  "Solved".  
- In In[22] you have this line: 
  
  ```
  ERROR! Session/line number was not unique in database. History logging moved to new    session 66 
  ```
  
  What is it?
     I'm trying to run this one, sometimes I get errors and when it runs it takes forever. I guess it is because of the issue of calling 4 different functions "init". We should be able to run your code with no errors. 
- How we know that this results are correct, do you have any source or problem for comparison/test?  

typos:

functions or the ----> functions of the 
And What ------------> And what

AkshayaSrivastava commented:
Thanks for all the feedback!

I'll get started on making those changes immediately. Do I have to do another pull request after each commit, or once I'm done making the changes?

labarba commented:
Any new commits get automatically added to this pull request.

AkshayaSrivastava commented:
ok. thanks!

AkshayaSrivastava commented:
I rewrote the assignment, focusing less on showing the 1st order neumann condition fails, and more on the second order scheme, I also fixed all of the errors that were appearing earlier, (the last commit message has more info on that), and I've moved the older versions into a separate folder. Please use V2 for grading. 

Merry Xmas to all and to all a good night!
